ecl_geometry/include/ecl/geometry/angle.hpp

```diff
@@ -38,7 +38,7 @@ namespace ecl {
  * @return T : the angle in degrees.
  */
 template <typename T>
-ecl_geometry_PUBLIC T radians_to_degrees(const T &radians, typename enable_if<ecl::is_float<T> >::type* dummy = 0) {
+T radians_to_degrees(const T &radians, typename enable_if<ecl::is_float<T> >::type* dummy = 0) {
```

Comment:
this would not be a problem. `ecl_geometry_PUBLIC` would be `ECL_HELPER_EXPORT` or `ECL_HELPER_IMPORT` (depending on the desired visibility). These 2 point to the same definition for gcc

related definition:
```c++
#ifdef ECL_HAS_SHARED_LIBS // ecl is being built around shared libraries
  #ifdef ecl_geometry_EXPORTS // we are building a shared lib/dll
    #define ecl_geometry_PUBLIC ECL_HELPER_EXPORT
  #else // we are using shared lib/dll
    #define ecl_geometry_PUBLIC ECL_HELPER_IMPORT
  #endif
  #define ecl_geometry_LOCAL ECL_HELPER_LOCAL
#else // ecl is being built around static libraries
  #define ecl_geometry_PUBLIC
  #define ecl_geometry_LOCAL
#endif
```

```c++
#if defined(ECL_IS_WIN32) || defined(ECL_IS_CYGWIN)
  #define ECL_HELPER_IMPORT __declspec(dllimport)
  #define ECL_HELPER_EXPORT __declspec(dllexport)
  #define ECL_HELPER_LOCAL
#else
  #if defined(ECL_IS_POSIX) && __GNUC__ >= 4
    #define ECL_HELPER_IMPORT __attribute__ ((visibility("default")))
    #define ECL_HELPER_EXPORT __attribute__ ((visibility("default")))
    #define ECL_HELPER_LOCAL  __attribute__ ((visibility("hidden")))
  #else
    #define ECL_HELPER_IMPORT
    #define ECL_HELPER_EXPORT
    #define ECL_HELPER_LOCAL
  #endif
#endif
```

Reply:
These symbols should be exported, so let's verify

Reply:
since these functions/classes already include implementation, `dllimport` cannot be used (otherwise would cause this error `error C2491: definition of dllimport function not allowed`)